cmd/ilm-list.go

```diff
@@ -0,0 +1,285 @@
+		ctx.Bool("minimum")}
+	foundSet := false
+	idx := 0
+	for range flags {
+		if foundSet && flags[idx] {
+			return true
+		} else if flags[idx] {
+			foundSet = true
+		}
+		idx++
+	}
+	return false
+}
+
+// checkILMListSyntax - validate arguments passed by a user
+func checkILMListSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelp(ctx, "list")
+		os.Exit(globalErrorExitStatus)
+	}
+	if invalidILMListFlagSet(ctx) {
+		fatalIf(probe.NewError(errors.New("Invalid input flag(s)")), "Only one show field flag allowed for list command. Refer mc "+ctx.Command.FullName()+" --help.")
+	}
+}
+
+const tableSeperator = "|"
+
+func getILMHeader(tbl *PrettyTable, alignedHdrLabels ...string) string {
+	if len(alignedHdrLabels) == 0 {
+		return ""
+	}
+	row := tbl.buildRow(alignedHdrLabels...)
+	header := console.Colorize(ilmThemeHeader, row+"\n")
+	lineRow := buildILMTableLineRow(alignedHdrLabels...)
+	row = tbl.buildRow(lineRow...)
+	row = console.Colorize(ilmThemeHeader, row+"\n")
+	header += row
+	return header
+}
+
+func buildILMTableLineRow(rowArr ...string) []string {
+	lineRowArr := make([]string, len(rowArr))
+	for index := 0; index < len(rowArr); index++ {
+		var tagBfr bytes.Buffer
+		for rowArrChars := 0; rowArrChars < len(rowArr[index]); rowArrChars++ {
+			tagBfr.WriteByte('-')
+		}
+		lineRowArr[index] = tagBfr.String()
+	}
+	return lineRowArr
+}
+
+func getILMRowsNoTags(tbl *PrettyTable, cellDataNoTags *[][]string) string {
+	if cellDataNoTags == nil || len(*cellDataNoTags) == 0 {
+		return ""
+	}
+	var rows string
+	for _, rowArr := range *cellDataNoTags {
+		var row string // Table row
+		// Build & print row
+		row = tbl.buildRow(rowArr...)
+		row = console.Colorize(ilmThemeRow, row)
+		rows += row + "\n"
+		lineRow := buildILMTableLineRow(rowArr...)
+		row = tbl.buildRow(lineRow...)
+		row = console.Colorize(ilmThemeRow, row)
+		rows += row + "\n"
+	}
+	return rows
+}
+
+func getILMRowsWithTags(tbl *PrettyTable, cellDataWithTags *[][]string, newRows map[string][]string) string {
+	if cellDataWithTags == nil || len(*cellDataWithTags) == 0 {
+		return ""
+	}
+	var rows string
+	for _, rowArr := range *cellDataWithTags {
+		if rowArr == nil {
+			continue
+		}
+		var row string // Table row
+		// Build & print row
+		row = tbl.buildRow(rowArr...)
+		row = console.Colorize(ilmThemeRow, row)
+		rows += row + "\n"
+		// Add the extra blank rows & tag value in the right column
+		if len(newRows) > 0 {
+			for index := 0; index < len(newRows); index++ {
+				newRow, ok := newRows[strings.TrimSpace(rowArr[0])+strconv.Itoa(index)]
+				if ok {
+					row = tbl.buildRow(newRow...)
+					row = console.Colorize(ilmThemeRow, row)
+					rows += row + "\n"
+				}
+			}
+		}
+		// Build & print the line row
+		lineRow := buildILMTableLineRow(rowArr...)
+		row = tbl.buildRow(lineRow...)
+		row = console.Colorize(ilmThemeRow, row)
+		rows += row + "\n"
+	}
+	return rows
+}
+
+func mainILMList(ctx *cli.Context) error {
```

Comment:
I noticed the following while testing this PR locally. Notice that the tags column in the tabular output seems truncated.
```➜  mc git:(ilm-table-display-2) ✗ ./mc ilm list --json play/kp-ilm
{
 "status": "success",
 "target": "play/kp-ilm",
 "ilm": {
  "Rules": [
   {
    "Expiration": {
     "Days": 200
    },
    "ID": "Devices",
    "Filter": {
     "And": {
      "Tags": [
       {
        "Key": "docformat",
        "Value": "docx"
       },
       {
        "Key": "plaintextformat",
        "Value": "txt"
       },
       {
        "Key": "exportFormat",
        "Value": "pdf"
       }
      ]
     }
    },
    "Status": "Enabled"
   }
  ]
 }
}
➜  mc git:(ilm-table-display-2) ✗ ./mc ilm list play/kp-ilm       
     ID     |  Prefix  |  Enabled   | Expiry |  Date/Days   |  Transition  |    Date/Days     |  Storage-Class   |       Tags       
------------|----------|------------|--------|--------------|--------------|------------------|------------------|------------------
 Devices    |          |    ✓       |  ✓     |  200 day(s)  |     ✗        |                  |                  | docformat:docx   
            |          |            |        |              |              |                  |                  | plaintextform..  
            |          |            |        |              |              |                  |                  | exportFormat:..  
------------|----------|------------|--------|--------------|--------------|------------------|------------------|------------------
```

Reply:
Yes that is by design. We want to restrict tags column to a certain width. The _..._ will indicate to the user that the tags are clipped/trimmed. The way PrettyTable is designed, we need to keep the column widths constant. Shall we change it when we move to console Table?